### PR TITLE
fix(dav): Remove shares of inexistent sharers

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Resolves: # 

## Summary

`occ dav:remove-invalid-shares` removes shares when the share recipient does not exist (anymore). This also adds a check for the sharers. If they don't exist, shares will be dropped.

## TODO

- [x] Do

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/Command/RemoveInvalidShares.php [hunk 1]
```diff
@@ -37,44 +38,95 @@
...
-			$p = $this->principalBackend->getPrincipalByPath($principaluri);
-			if ($p === null) {
-				$this->deleteSharesForPrincipal($principaluri);
+			if (!isset($this->cache[$principaluri])) {
+				$exists = $this->cache[$principaluri] = $this->principalBackend->getPrincipalByPath($principaluri) !== null;
+			} else {
+				$exists = $this->cache[$principaluri];
+			}
+			if (!$exists) {
+				$output->writeln("Principal URI $principaluri is an invalid share recipient", OutputInterface::VERBOSITY_VERBOSE);
```
Comment (tcitworld) [inline apps/dav/lib/Command/RemoveInvalidShares.php:73]: If the `dry-run` option is given it should probably be set to `VERBOSITY_NORMAL`.

**Review by tcitworld — COMMENTED** [review]
Is using a cache worth it rather than just using the class properties to keep track of key/values booleans ?

**Comment by blizzz** [comment]
Please don't forget to set the milestone when opening a PR :smiley:

**Comment by ChristophWurst** [comment]
This was not needed in the end

**Pull request closed without merging**